src/test/java/org/jenkinsci/plugins/gitclient/GitAPITestCase.java

```diff
@@ -514,6 +514,132 @@ public void test_setCommitter() throws Exception {
         assertTrue("Wrong committer in " + revision, revision.get(3).startsWith("committer " + committerName + " <" + committerEmail + "> "));
     }
 
+    /** Clone arguments include:
+     *   repositoryName(String) - if omitted, CliGit does not set a remote repo name
+     *   shallow() - no relevant assertion of success or failure of this argument
+     *   shared() - not implemented on CliGit, not verified on JGit
+     *   reference() - implemented on JGit, not verified on either JGit or CliGit
+     *
+     * CliGit and JGit both require the w.git.checkout() call
+     * otherwise no branch is checked out. That is different than the
+     * command line git program, but consistent within the git API.
+     */
+    public void test_clone() throws Exception
+    {
+        cloneTimeout = 1 + random.nextInt(60 * 24);
+        w.git.clone_().timeout(cloneTimeout).url(localMirror()).repositoryName("origin").execute();
+        createRevParseBranch(); // Verify JENKINS-32258 is fixed
+        w.git.checkout().ref("origin/master").branch("master").execute();
+        check_remote_url("origin");
+        assertBranchesExist(w.git.getBranches(), "master");
+        final String alternates = ".git" + File.separator + "objects" + File.separator + "info" + File.separator + "alternates";
+        assertFalse("Alternates file found: " + alternates, w.exists(alternates));
+        assertFalse("Unexpected shallow clone", w.cgit().isShallowRepository());
+    }
+
+    public void test_checkout_exception() throws Exception {
+        w.git.clone_().url(localMirror()).repositoryName("origin").execute();
+        createRevParseBranch();
+        w.git.checkout().ref("origin/master").branch("master").execute();
+        final String SHA1 = "feedbeefbeaded";
+        try {
+            w.git.checkout().ref(SHA1).branch("master").execute();
+            fail("Expected checkout exception not thrown");
+        } catch (GitException ge) {
+            assertEquals("Could not checkout master with start point " + SHA1, ge.getMessage());
+        }
+    }
+
+    public void test_clone_repositoryName() throws IOException, InterruptedException
+    {
+        w.git.clone_().url(localMirror()).repositoryName("upstream").execute();
+        w.git.checkout().ref("upstream/master").branch("master").execute();
+        check_remote_url("upstream");
+        assertBranchesExist(w.git.getBranches(), "master");
+        final String alternates = ".git" + File.separator + "objects" + File.separator + "info" + File.separator + "alternates";
+        assertFalse("Alternates file found: " + alternates, w.exists(alternates));
+    }
+
+    public void test_clone_longPath() throws IOException, InterruptedException
+    {
+        w.git.clone_().url(localMirror()).repositoryName("upstream").longPath(true).execute();
+        assertThat("true", is(getConfigValue(w.repo, "core.longpaths")));
```

Comment:
Hamcrest assertion message will display the right hand value as the expected value and the left hand value as the actual value. 
```suggestion
        assertThat(getConfigValue(w.repo, "core.longpaths"), is("true"));
```

Reply:
This test should move to one of the GitClient*Test classes so that it tests all the implementations and is expressed as JUnit 4 test.  Then you can rest GitAPITestCase to match the contents of the master branch.

Reply:
I would also like a test that performs a clone into a directory with an absolute path longer than 260 characters.  The test for the config value is good and necessary, but a test that things are well behaved is even better.